app/controllers/concerns/dcv/catalog/csv_download_behavior.rb

```diff
@@ -21,16 +21,15 @@ def stream_csv_response_for_search_results
     per_page = 2000
     fl = field_keys_to_labels.keys.join(',') # only retrieve the fields we care about. much faster than asking for all fields.
     begin
-      while (
+      while @document_list&.[](1).present? do
```

Comment:
there needs to be an integration test to ensure this paginates correctly while not doubling the first requrst